ims/einstein/operations.py

```diff
@@ -217,11 +217,16 @@ def shutdown(self):
 
     # Provisions from HaaS and Boots the given node with given image
     @log
-    def provision(self, node_name, img_name, network, nic):
+    def provision(self, node_name, img_name, network, nic, hil_flag):
         try:
+            hil_bool = bool(hil_flag == "True")
```

Comment:
I would not recommend that here, as I would like to have semantic transparency in the redesign with type-safety via message passing where each field is typed.  By the way, there is no json here - it's just unicode for each parameter.  We need to redesign BMI properly from the ground up, and to infer types in the current design is just work that we can put in the redesign.

This is for the cloud project so that they have their paper published in time.